lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -892,6 +896,128 @@ class SubtractOpConversionPattern
   }
 };
 
+class AllReduceOpConversionPattern
+    : public OpConversionPattern<ttir::AllReduceOp> {
+public:
+  using OpConversionPattern<ttir::AllReduceOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(ttir::AllReduceOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    RankedTensorType inputType =
+        mlir::cast<RankedTensorType>(adaptor.getInputs().front().getType());
+    SmallVector<int64_t> inputTypeShape(inputType.getShape());
+    auto replicaGroupsShape = adaptor.getReplicaGroups().getType().getShape();
+    size_t scatter_dim = adaptor.getDim();
+    int32_t scatter_num =
+        replicaGroupsShape[scatter_dim % replicaGroupsShape.size()];
+    auto device = getOrInsertDevice(rewriter, op);
+
+    // This lowering is supposed to be one to one mapping from ttir all_reduce
+    // to ttnn all_reduce. However, the all_reduce ops are broken down into
+    // reduce_scatter and all_gather ops because current support of all_reduce
+    // in TTNN is not stable.
+
+    // The reduce_scatter op in TTNN currently does not support two dimensional
+    // tensor correctly. As a temporary workaround, we insert reshape ops front
+    // and back to make the tensor as four dimensional tensor.
+
+    // TODO(wooseoklee): Once it supports two dimensional tensor, we can remove
```

Comment:
Can we create an issue for this, refer to it here so that we can track it in GH...

Reply:
This issue tracks the two dim issue in reduce scatter, https://github.com/tenstorrent/tt-metal/issues/15010.

Reply:
@wooseokTT can we add a link as part of the code comment?